### DISCUSSION TITLE
Failed validation re-encoding

### DISCUSSION THREAD

**Discussion by billyors** [ptr727/PlexCleaner#165] [Q&A] [answered]
Hello!

Is there a clever way to make files that fail stream validation convert to their original encoding?

i.e. My failed file is x265 (hevc), by default PlexCleaner will now re-encode to x264. 

I think the logic makes sense because it's what I'm telling it to do by default, but I have folders with mixed encodings 😨 . I haven't found a way to make it identify if x265, re-encode in x265 (upon vaildation failure) and if x264, re-encode in x264.

Many thanks.

**Comment by ptr727** [comment, marked as answer]
No such ability, repair using encoding uses the encoding settings.  
Out of interest, why does the original format matter?

**Reply by billyors** [reply]
Thanks for the response. Kudos to you and this project btw, it's been a gem 💎 .

From my limited understanding of encoding, the raw file is the only thing that should be the source of encoding. If you encode an already compressed x265 to x264 then you find a loss in quality and inflate file sizes. Vise-versa, if x264 is converted to x265 then it's considered the same lane of bad practice, although you'll see the benefit of smaller file sizes in this case.

I could be wrong, the internet may be lying and this isn't as bad as I think it is for playback to convert existing encodings.

**Reply by ptr727** [reply]
Any form of re-encoding using any codec will result in a loss of quality (excluding lossless that will dramatically increase the output size), this is unavoidable.

Consider the typical pipeline: original raw stream (you do not have access to this) -> encoded stream (your media file) -> decoder -> raw uncompressed stream (in-memory) -> encoder -> encoded stream (repaired file).

There is no benefit to using the same codec as originally encoded, one may argue that using a better or actively maintained codec for re-encoding may be preferred.

**Reply by billyors** [reply]
Thank you for the breakdown. Albeit I'm still grasping the fundamentals here, this helps clarify the answer to my question about using the original codec.
